### PR TITLE
District Champs code

### PR DESCRIPTION
**Pull request by TheLocust3**

### CODE REVIEW

**Comment by TheLocust3** [comment]
There are some weird limits to the hood and turntable. They are controlled by each other and are somewhat weird.
There was also lots of tuning
Probably some other stuff but I don't remember it right now

File: src/org/robockets/stronghold/robot/commands/SetPID.java [hunk 1]
```diff
@@ -29,15 +29,15 @@ public SetPID(String name, PIDController controller, double P, double I, double
     protected void initialize() {
     	//controller.disable();
     	
-    	SmartDashboard.putNumber(name + " p", SmartDashboard.getNumber(name + " p", controller.getP()));
-    	SmartDashboard.putNumber(name + " i", SmartDashboard.getNumber(name + " i", controller.getI()));
-    	SmartDashboard.putNumber(name + " d", SmartDashboard.getNumber(name + " d", controller.getD()));
+    	SmartDashboard.putNumber(name + " p", SmartDashboard.getNumber(name + " p", controller.getP() * 1000.0));
```
Comment (bshin100) [inline src/org/robockets/stronghold/robot/commands/SetPID.java]: What's the reason for multiplying/dividing by 1000?

File: src/org/robockets/stronghold/robot/hood/Hood.java [hunk 2]
```diff
@@ -24,26 +36,57 @@ public Hood(){
...
+    	
+    	if (angle < HOOD_MIN) {
+    		Robot.turntable.positiveLimit = Robot.turntable.OPEN_SIDE_LIMIT;
+    		Robot.turntable.negativeLimit = Robot.turntable.MIDPOINT;
+    	} else {
+    		Robot.turntable.positiveLimit = Robot.turntable.OPEN_SIDE_LIMIT;
+    		Robot.turntable.negativeLimit = Robot.turntable.MOTOR_SIDE_LIMIT;
+    	}
+    	
+		setSetpoint(angle);
```
Comment (bshin100) [inline src/org/robockets/stronghold/robot/hood/Hood.java]: What is this setting the setpoint of?

**Comment by bshin100** [comment]
Seems about right, make sure everything works with a robot. :+1:

File: src/org/robockets/stronghold/robot/highgoalshooter/TurnUntilTarget.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
+package org.robockets.stronghold.robot.highgoalshooter;
+
+import org.robockets.stronghold.robot.Robot;
+import org.robockets.stronghold.robot.turntable.HorizontalAlign;
+
+import edu.wpi.first.wpilibj.command.Command;
+import edu.wpi.first.wpilibj.networktables.NetworkTable;
+
+/**
+ *
```
Comment (TheLocust3) [inline src/org/robockets/stronghold/robot/highgoalshooter/TurnUntilTarget.java]: fill me

File: src/org/robockets/stronghold/robot/highgoalshooter/TurnUntilTarget.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+    // Called just before this Command runs the first time
+    protected void initialize() {
+    	visionTable = NetworkTable.getTable("vision");
+    }
+
+    // Called repeatedly when this Command is scheduled to run
+    protected void execute() {
+    	Robot.driveTrain.driveArcade(0, speed);
+    	System.out.println("TEST!");
```
Comment (TheLocust3) [inline src/org/robockets/stronghold/robot/highgoalshooter/TurnUntilTarget.java]: Remove this

File: src/org/robockets/stronghold/robot/turntable/Turntable.java [hunk 5]
```diff
@@ -37,7 +47,24 @@ public double getAngle() {
     }
     
     public void setAngle(double angle) {
-    	//pidController.setPID(SmartDashboard.getNumber("P", 0), SmartDashboard.getNumber("I", 0), SmartDashboard.getNumber("D", 0));
+    	//pidController.setSetpoint(SmartDashboard.getNumber("New Turntable"));
+    	
+    	atLimit = false;
+    	// Limit the turntable turning to -45 to 45
```
Comment (TheLocust3) [inline src/org/robockets/stronghold/robot/turntable/Turntable.java]: Probably should remove this misleading comment

File: src/org/robockets/stronghold/robot/drivetrain/Joyride.java [hunk 6]
```diff
@@ -25,6 +27,9 @@ protected void initialize() {
 
     // Called repeatedly when this Command is scheduled to run
     protected void execute() {
+    	counter++;
+    	
+    	System.out.println(counter);
```
Comment (TheLocust3) [inline src/org/robockets/stronghold/robot/drivetrain/Joyride.java]: We can remove this counter now

File: src/org/robockets/stronghold/robot/commands/Autonomous.java [hunk 7]
```diff
@@ -23,35 +26,40 @@
...
     public  Autonomous(double autoNumber, double defense) {
-    	//addParallel(new SetShooterFlipper(0.1));
-    	addParallel(new SetShooterFlipper(0.5));
+    	addParallel(new SetShooterFlipper(0.1));
+    	//addParallel(new SetShooterFlipper(0.5));
 
-    	if (autoNumber == 1) { // Lowbar + Porticullis, no shoot
-    		addSequential(new WaitCommand(3));
-    		addSequential(new AutoLowBar(Direction.FORWARD));
+    	if (autoNumber == 1) { // Lowbar, no shoot
```
Comment (simon-andrews) [inline src/org/robockets/stronghold/robot/commands/Autonomous.java]: Can't this whole piece be done as a switch?

**Comment by TheLocust3** [comment]
Merging

**Pull request merged**